env/io_posix.cc

```diff
@@ -711,13 +711,30 @@ IOStatus PosixRandomAccessFile::MultiRead(FSReadRequest* reqs,
           // comment
           // https://github.com/facebook/rocksdb/pull/6441#issuecomment-589843435
           // Fall back to pread in this case.
-          Slice tmp_slice;
-          req->status =
-              Read(req->offset + req_wrap->finished_len,
-                   req->len - req_wrap->finished_len, options, &tmp_slice,
-                   req->scratch + req_wrap->finished_len, dbg);
-          req->result =
-              Slice(req->scratch, req_wrap->finished_len + tmp_slice.size());
+          if (use_direct_io()) {
+            if (req_wrap->finished_len %
+                    static_cast<size_t>(GetRequiredBufferAlignment()) !=
+                0) {
+              // Bytes reads don't fill sectors. Should only happen at the end
+              // of the file.
+              req->result = Slice(req->scratch, req_wrap->finished_len);
+              req->status = IOStatus::OK();
+            } else {
+              // Returns partial result, need to re-read with pread.
+              // Since offset must be aligned,
+              // read from the original offset again.
+              req->status = Read(req->offset, req->len, options, &req->result,
```

Comment:
In this else case, ```finished_len``` would be aligned right? So it should be ok to call Read with ```req->offset + req_wrap->finished_len```? Not a big deal, but I think it would make this whole block simpler.